### PR TITLE
[SPARK-8705][WebUI]Don't display rects when totalExecutionTime is 0

### PR DESCRIPTION
**Pull request by zsxwing**
Because `System.currentTimeMillis()` is not accurate for tasks that only need several milliseconds, sometimes `totalExecutionTime` in `makeTimeline` will be 0. If `totalExecutionTime` is 0, there will the following error in the console.

screen shot 2015-06-29 at 7 08 55 pm

This PR fixes it by using an empty svg tag when `totalExecutionTime` is 0. This is a screenshot for a task that its totalExecutionTime is 0 after fixing it.

screen shot 2015-06-30 at 12 26 52 am

### CODE REVIEW

**Comment by sarutak** [comment]
LGTM, but just in case, could you show a screen shot in the case of `totalExecutionTime == 0` ?

**Comment by zsxwing** [comment]
> could you show a screen shot in the case of totalExecutionTime == 0 ?

I have already put a screenshot in the PR description.

**Comment by sarutak** [comment]
Ah, that is after patch applied. O.K, I'll merge once the test finish successfully.

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala [hunk 1]
```diff
@@ -570,6 +570,35 @@ private[ui] class StagePage(parent: StagesTab) extends WebUIPage("stage") {
...
+
+        val svgTag =
+          if (totalExecutionTime == 0) {
+            """<svg class="task-assignment-timeline-duration-bar"></svg>"""
+          }
+          else {
+           s"""<svg class="task-assignment-timeline-duration-bar">
+                 |<rect class="scheduler-delay-proportion"
+                   |x="$schedulerDelayProportionPos%" y="0px" height="26px"
+                   |width="$schedulerDelayProportion%""></rect>
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: is there an extra `"` here?

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala [hunk 2]
```diff
@@ -570,6 +570,35 @@ private[ui] class StagePage(parent: StagesTab) extends WebUIPage("stage") {
 
         val index = taskInfo.index
         val attempt = taskInfo.attempt
+
+        val svgTag =
+          if (totalExecutionTime == 0) {
+            """<svg class="task-assignment-timeline-duration-bar"></svg>"""
+          }
+          else {
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: this should go on the previous line

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala [hunk 3]
```diff
@@ -570,6 +570,35 @@ private[ui] class StagePage(parent: StagesTab) extends WebUIPage("stage") {
 
         val index = taskInfo.index
         val attempt = taskInfo.attempt
+
+        val svgTag =
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: can you add a short comment here that links against the JIRA:
```
SPARK-8705: Avoid invalid attribute error in JavaScript if execution time is 0
```

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala (hunk 1 above)
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: Good catch

**Comment by zsxwing** [comment]
Addressed comments.

**Comment by sarutak** [comment]
I think, we can merge this. @andrewor14 do you think this is ready to merge?

**Comment by andrewor14** [comment]
Looks great. I'm merging this into master and 1.4.

**Comment by andrewor14** [comment]
@zsxwing looks like this doesn't merge cleanly into branch-1.4. Can you open a new patch against that branch?

**Comment by zsxwing** [comment]
@andrewor14 this is because #7082 is not merged into branch-1.4. Do you think if it's worth to merge into branch-1.4?

**Comment by andrewor14** [comment]
Oh, then it's OK

**Pull request closed without merging**